bundles/sirix-core/src/test/java/org/sirix/service/json/serialize/JsonSerializerTest.java

```diff
@@ -394,6 +394,46 @@ public void testJsonDocumentWithMaxLevel() throws IOException {
     }
   }
 
+  @Test
+  public void testJsonDocumentWithMaxLevelAndNumberOfNodes() throws IOException {
+    JsonTestHelper.createTestDocument();
+
+    final var database = JsonTestHelper.getDatabase(PATHS.PATH1.getFile());
+    try (final var manager = database.openResourceManager(JsonTestHelper.RESOURCE)) {
+      try (final Writer writer = new StringWriter()) {
+        final var serializer = new JsonSerializer.Builder(manager, writer).maxLevel(2).numberOfNodes(3).build();
+        serializer.call();
+
+        final var expected = "{\"foo\":[]}";
+        assertEquals(expected, writer.toString());
+      }
+
+      try (final Writer writer = new StringWriter()) {
+        final var serializer = new JsonSerializer.Builder(manager, writer).maxLevel(2).numberOfNodes(4).build();
+        serializer.call();
+
+        final var expected = "{\"foo\":[]}";
+        assertEquals(expected, writer.toString());
+      }
+
+      try (final Writer writer = new StringWriter()) {
+        final var serializer = new JsonSerializer.Builder(manager, writer).maxLevel(2).numberOfNodes(5).build();
+        serializer.call();
+
+        final var expected = "{\"foo\":[],\"bar\":{}}";
+        assertEquals(expected, writer.toString());
+      }
+
+      try (final Writer writer = new StringWriter()) {
+        final var serializer = new JsonSerializer.Builder(manager, writer).maxLevel(2).numberOfNodes(6).build();
```

Comment:
This seems redundant, it would be clean if we write another method like
`private String getSerializedString(int maxLevel, int numberOfNodes)`